core/src/main/web/app/beaker.js

```diff
@@ -272,18 +272,21 @@
       }
     });
 
-    beaker.run(function($location, $route, $document, bkUtils, bkCoreManager, bkHelper) {
+    beaker.run(function($rootScope, $location, $route, $document, bkUtils, bkCoreManager, bkHelper) {
       var user;
       var lastAction = new Date();
       var beakerRootOp = {
         gotoControlPanel: function() {
-          return $location.path("/control").search({});
+          var ret = $location.path("/control").search({});
+          if (bkUtils.isElectron) {
+            $rootScope.$apply();
```

Comment:
looks like we should be doing this inside of an evalAsync

https://github.com/angular/angular.js/issues/9607#issuecomment-59087310